### PR TITLE
Prepare account deletion

### LINKED ISSUE #51: Add the ability to delete an account
We need to add the ability to delete an account by the user. After removing him, all polls created by him should also be deleted.

This option should be in the 'Setting' tab in the drop-down menu. If the bookmark has not been created in the meantime, it should be added as part of this task

image

### PR DESCRIPTION
**Pull request by xdk78**
Resolve #51

### CODE REVIEW

File: src/layout/Navigation/Navigation.tsx [hunk 1]
```diff
@@ -76,7 +76,14 @@ function Navigation() {
...
                   <Menu.Items className="absolute right-0 mt-2 bg-white rounded-md divide-y divide-zinc-100 focus:outline-none ring-1 ring-black ring-opacity-5 shadow-lg origin-top-right">
-                    <div className="flex justify-end p-1">
+                    <div className="flex flex-col justify-end p-1">
+                      <Menu.Item>
+                        <Link href={'/settings'} passHref>
+                          <ButtonLink variant={ButtonVariant.FLAT}>
+                            Settings
+                          </ButtonLink>
+                        </Link>
+                      </Menu.Item>
```
Comment (Ryczko) [inline src/layout/Navigation/Navigation.tsx:86]: What about burger menu?

File: src/layout/Navigation/Navigation.tsx [hunk 2]
```diff
@@ -76,7 +76,14 @@ function Navigation() {
                   leaveTo="transform opacity-0 scale-95"
                 >
                   <Menu.Items className="absolute right-0 mt-2 bg-white rounded-md divide-y divide-zinc-100 focus:outline-none ring-1 ring-black ring-opacity-5 shadow-lg origin-top-right">
-                    <div className="flex justify-end p-1">
+                    <div className="flex flex-col justify-end p-1">
+                      <Menu.Item>
+                        <Link href={'/settings'} passHref>
+                          <ButtonLink variant={ButtonVariant.FLAT}>
+                            Settings
```
[the remark this answers is not in the document]
Comment (xdk78) [reply src/layout/Navigation/Navigation.tsx:83]: what is cog?

[the remark this answers is not in the document]
Comment (Ryczko) [reply src/layout/Navigation/Navigation.tsx:83]: image

**Review by Ryczko — CHANGES_REQUESTED** [review]
3 problems has been found:

- [ ] Sometimes on my surveys page when you click delete button and then cancel on modal, all page is blocked (nothing can be clicked)

[screen-capture (1).webm](https://user-images.githubusercontent.com/51440879/226172975-e68925af-2eb8-4447-9015-c3b75a7e791c.webm)


- [ ] 'Delete my account' button jumps when modal is opened.

[screen-capture (2).webm](https://user-images.githubusercontent.com/51440879/226173066-36f3cddd-007f-4448-b9bd-70f99ff3e688.webm)


- [ ] When settings button is hover it should takes all width on mobile and desktop:
image
image

File: src/pages/settings/index.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    openDeleteModal,
+    closeDeleteModal,
+    handleOnAccountDelete,
+  } = useSettingsManager();
+
+  return (
+    <>
+      <Head>
+        <title>Settings</title>
+      </Head>
```
Comment (pmatyjasik) [inline src/pages/settings/index.tsx:28]: You should add meta tag in Head component (Look on homepage file)

File: src/pages/settings/index.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+          }
+        />
+        {loading && (
+          <div className="text-sm text-center text-zinc-600">Loading...</div>
+        )}
+        {error && (
+          <div className="text-sm text-center text-red-600">
+            Error: {error.message}
+          </div>
+        )}
```
Comment (pmatyjasik) [inline src/pages/settings/index.tsx:83]: Remove that error

Comment (xdk78) [reply src/pages/settings/index.tsx:83]: why?

Comment (xdk78) [reply src/pages/settings/index.tsx]: ok

**Review by Ryczko — APPROVED** [review]

**Pull request merged**